From d53ca143e45f4bd91a4a1dddc316556f43279f46 Mon Sep 17 00:00:00 2001
From: Seth Shelnutt <Shelnutt2@gmail.com>
Date: Thu, 28 Apr 2022 10:01:59 -0400
Subject: [PATCH] Fix comparison error for pandas dataframe dtype

The dtype passed in is likely to be of numpy.dtype when the dtype comes
directly from pandas. This means we need to check the type before
comparing to the string "ascii".
---
 tiledb/dataframe_.py | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/tiledb/dataframe_.py b/tiledb/dataframe_.py
index c50c1c7..80c1add 100644
--- a/tiledb/dataframe_.py
+++ b/tiledb/dataframe_.py
@@ -112,7 +112,9 @@ class ColumnInfo:
     def from_dtype(cls, dtype, varlen_types=()):
         from pandas.api import types as pd_types
 
-        if dtype == "ascii":
+        if (type(dtype) == str and dtype == "ascii") or (
+            type(dtype) == np.dtype and str(dtype) == "ascii"
+        ):
             return cls("ascii", var=True)
 
         dtype = pd_types.pandas_dtype(dtype)
-- 
2.31.1

